connect-integrations.md

```diff
@@ -0,0 +1,194 @@
+   to consuming connect. They may not consume for a long time, then have a small
+   burst of messages, and then go back to not consuming for a while. They may
+   have such a small amount of traffic that it takes several hours to fill the
+   GZIP buffer (Connect is gzipped by default, because it is optimized for
+   apps on the larger side).
+
+   Moreover, once you get enough services running, you're basically guaranteed
+   that a couple are always going to be in a wonkey state for some amount of
+   time, whether due to network unreliability, unexpected thirdparty errors (I'm
+   looking at you AWS S3: `We encountered an internal error. Please try again.
+   (Service: null; Status Code: 0; Error Code: InternalError; Request ID:
+   00090C0A4CA19DB0)`, or customer misconfiguration. Rather than try to
+   categorize all the problems an integration might come accross, it's better to
+   just to record that the service fulfilling integration #2 for a given app has
+   encontered an exception, and report the exception to sentry. 
+
+   The converse of this rule, "If a lot of services are experiencing a problem,
+   then we have a problem" means that we need something that periodically scans
+   the table where exceptions encountered by tasks are stored, and makes a
+   determination about whether or not the system is healthy. If a lot of
+   services are in a problem state, it's certain that something somehwere is
+   wrong and requires human intervention.
+
+3. If a service consuming for an app with a large audience has 
+   a problem, then we have a problem.
+
+   The above rule almost always holds true, unless the problematic integration
+   is particularly reliable or important to the success of the product.
+
+   Large apps fulfill this criteria:
+
+   - Connect is optimized for large apps, a large app having problems is
+     likely to be a real problem. 
+   - Large apps make up a substantial portion of the income the project
+     generates, and they have leverage to demand discounts if they
+     perceive the service to be unreliable.
+   - Large apps are unlikely to configure their integration incorrectly, since
+     if they do they are likely to complain (since they expect a higher quality
+     of functionality given the cost of their subscription).
+
+The controller service scans the dynamo table, and keeps counts of how many 
+integrations have encountered exceptions, how many are lagged, and how many
+are starting up very frequently. It sends these counts to a 
+prometheus AlertManager, which notifies us if a sufficient proportion of them 
+are in alerting territory. 
+
+The goal is to never page unless the problem is real, but also to make sure we
+know when something is wrong. I think this balance does that-- intermittent
+issues are surfaced by sentry, on a timeline that is convenient for the
+developer. Major issues are surfaced immediately by our system level alerting.
```

Comment:
I think a strong conclusion could be about the ease of writing configurations, but the biggest amount of work for this type of system is the monitoring and alerting.